test/e2e/collection/namespace_filtering_test.go

```diff
@@ -0,0 +1,188 @@
+		}
+
+	})
+
+	GetNamespace := func(namespace string) (string, error) {
+		options := metav1.ListOptions{
+			LabelSelector: "component=elasticsearch",
+		}
+		app_index := "app-000001"
+		pods, err := e2e.KubeClient.CoreV1().Pods(constants.OpenshiftNS).List(context.TODO(), options)
+		if err != nil {
+			return "", errors.New("Getting pods error")
+		}
+		if len(pods.Items) == 0 {
+			return "", errors.New("No pods found for elasticsearch")
+		}
+		log.V(3).Info("Pod ", "PodName", pods.Items[0].Name)
+		query := "--query=" + app_index + "/_search?size=10000"
+		stdout, err := e2e.PodExec(constants.OpenshiftNS, pods.Items[0].Name, "elasticsearch", []string{"es_util", query})
+		if err != nil {
+			return "", errors.New("Pods exec error")
+		}
+		// Get total hits
+		var jsonMap map[string]interface{}
+		if err := json.Unmarshal([]byte(stdout), &jsonMap); err != nil {
+			return "", errors.New("json unmarshall error")
+		}
+		total := jsonMap["hits"].(map[string]interface{})["total"]
+		val := int(total.(float64))
+
+		// Parse each document and compare namespace value
+		if val > 0 {
+			docs := jsonMap["hits"].(map[string]interface{})["hits"].([]interface{})
+			for _, document := range docs {
+				namespace_value := document.(map[string]interface{})["_source"].(map[string]interface{})["kubernetes"].(map[string]interface{})["namespace_name"].(string)
+				if namespace_value != namespace {
+					return namespace_value, nil
+				}
+			}
+		}
+		return namespace, nil
+	}
+
+	AssertBehaviours := func() {
+		It("should send logs from one namespace only", func() {
+			Expect(e2e.LogStores[elasticsearch.GetName()].HasApplicationLogs(framework.DefaultWaitForLogsTimeout)).To(BeTrue(), "Expected to find stored application logs")
+
+			namespace, err := GetNamespace(appNamespace1)
+			Expect(err).To(BeNil(), fmt.Sprintf("Error fetching logs: %v", err))
+			Expect(namespace).To(Equal(appNamespace1))
+		})
+	}
+
+	Describe("for fluentd collector", func() {
+		BeforeEach(func() {
+			cr := helpers.NewClusterLogging(helpers.ComponentTypeCollectorFluentd)
+			if err := e2e.CreateClusterLogging(cr); err != nil {
+				Fail(fmt.Sprintf("Unable to create an instance of cluster logging: %v", err))
+			}
+			if err := e2e.WaitFor(helpers.ComponentTypeCollectorFluentd); err != nil {
+				Fail(fmt.Sprintf("Failed waiting for component %s to be ready: %v", helpers.ComponentTypeCollectorFluentd, err))
+			}
+		})
+		AssertBehaviours()
+	})
+
+	Describe("for vector collector", func() {
+		BeforeEach(func() {
+			cr := helpers.NewClusterLogging(helpers.ComponentTypeCollectorVector)
+			if err := e2e.CreateClusterLogging(cr); err != nil {
+				Fail(fmt.Sprintf("Unable to create an instance of cluster logging: %v", err))
+			}
+			if err := e2e.WaitFor(helpers.ComponentTypeCollectorVector); err != nil {
+				Fail(fmt.Sprintf("Failed waiting for component %s to be ready: %v", helpers.ComponentTypeCollectorVector, err))
+			}
+		})
+		AssertBehaviours()
+	})
```

Comment:
is it possible to write this as a for loop? better if we can, and then update other tests also.

Reply:
Agree  here and with @alanconway .  we should convert this to a DescribeTable